File: internal/domain/team.go
package domain

type Team struct {
	BaseModel
	Name        string   `json:"name"`
	Description string   `json:"description"`
	Members     []Player `json:"members" gorm:"many2many:team_members;"`
	OwnerID     string   `json:"ownerId" gorm:"index"`
}

type TeamMember struct {
	TeamID   uint   `json:"teamId" gorm:"primaryKey"`
	PlayerID uint   `json:"playerId" gorm:"primaryKey"`
	Role     string `json:"role"`
}

func NewTeam(name, description, ownerID string) *Team {
	team := &Team{
		Name:        name,
		Description: description,
		OwnerID:     ownerID,
	}
	return team
}

func (t *Team) AddMember(player Player, role string) {
	t.Members = append(t.Members, player)
}

func (t *Team) RemoveMember(playerID uint) {
	var updatedMembers []Player
	for _, member := range t.Members {
		if member.ID != playerID {
			updatedMembers = append(updatedMembers, member)
		}
	}
	t.Members = updatedMembers
}

File: frontend/src/types/team.ts
export interface Team {
    id: number;
    name: string;
    description: string;
    ownerId: string;
    tenantId: number;
    createdAt: string;
    updatedAt: string;
    members: Player[];
    bookings: Booking[];
    costs: Cost[];
}

export interface Player {
    id: number;
    name: string;
    email: string;
    phone: string;
    role: string;
}

export interface Booking {
    id: number;
    teamId: number;
    courtId: number;
    startTime: string;
    endTime: string;
    status: 'pending' | 'confirmed' | 'cancelled';
    totalCost: number;
    description: string;
}

export interface Cost {
    id: number;
    teamId: number;
    amount: number;
    description: string;
    date: string;
    category: string;
}

export interface TeamMember {
    teamId: number;
    playerId: number;
    role: string;
} 